packages/creator-hub/main/src/modules/bin.ts

```diff
@@ -316,13 +317,79 @@ export async function dclDeepLink(deepLink: string) {
 
 export async function code(_path: string) {
   const normalizedPath = path.normalize(_path);
+  log.info('Opening project at path:', normalizedPath);
+
+  const config = await getConfig();
+  const editors = (await config.get('editors')) || [];
+  const defaultEditor = editors.find(editor => editor.isDefault);
+
+  log.info('Available editors:', editors);
+  log.info('Default editor:', defaultEditor);
+
   try {
-    await exec(`code "${normalizedPath}"`, { env: { ...process.env, PATH: getPath() } });
-    await track('Open Code', undefined);
-  } catch (_) {
-    const error = await shell.openPath(normalizedPath);
-    if (error) {
-      throw new Error(error);
+    if (defaultEditor) {
+      log.info('Opening with default editor:', defaultEditor.name, 'at path:', defaultEditor.path);
+      let command: string;
+      if (process.platform === 'darwin') {
+        const macosPath = path.join(defaultEditor.path, 'Contents', 'MacOS');
+        try {
+          const files = await fs.readdir(macosPath);
+
+          const executableFiles = await Promise.all(
+            files.map(async file => {
+              const filePath = path.join(macosPath, file);
+              try {
+                const stats = await fs.stat(filePath);
+
+                return stats.isFile() && stats.mode & 0o111 ? file : null;
+              } catch (error) {
+                return null;
+              }
+            }),
+          ).then(results => results.filter((file): file is string => file !== null));
```

Comment:
`.then` will not block the execution of the next line, so it can end up in unexpected race conditions

is not a good practice to mix `.then` with `await`.